### DISCUSSION TITLE
Issue with cider-eval-list-at-point (?)

### DISCUSSION THREAD

**Discussion by paulsc** [clojure-emacs/cider#3271] [General]
Hi all, I just posted this on the Discord as well, I'm not sure what the right channel is. 

cider-eval-list-at-point seems to behave oddly. Running it with the point at this position:

<img width="342" alt="Screenshot 2022-11-17 at 3 25 23 PM" src="https://user-images.githubusercontent.com/41470/202472038-ed04257c-0f67-462d-b01e-c94e748fa85f.png">

evaluates the (println ...) call instead of evaluating (* 6 2) 

Am I missing something or is this a bug ? It seems like substracting 1 to the cadr of the bounds in cider-eval-list-at-point fixes this. I can submit a patch if this is indeed a bug.

Thanks!

**Comment by bo-tato** [comment]
are you using evil mode? for me it behaves like that only while in evil mode, which of course I want to use so I'm using:
```
(defun my/cider-eval-list-at-point ()
  "eval within the surrounding parens"
  (interactive)
  (save-excursion
    (unless (eq ?\) (char-after))
      (evil-next-close-paren))
    (call-interactively #'cider-eval-last-sexp))
```
see https://github.com/clojure-emacs/cider/issues/3318
